### PR TITLE
Adding documenation for pull request review standards

### PR DESCRIPTION
**Pull request by maxinelasp**
# Change Summary
Adding a document to cover review standards

## New Files

- review-standards.rst

### CODE REVIEW

File: docs/source/development/style-guide/review-standards.rst
Comment (sdhoyt) [inline docs/source/development/style-guide/review-standards.rst]: We can all discuss this, but I think if the PR is an implementation for a level 5 requirement an issue needs to be created and linked to the PR. So there's some cases where just opening a PR without an issue and explaining in the PR is fine, but  I think a level 5 implementation or redesign of any kind should be a special case where an issue is required so we can easily keep track of the history of each level 5 by their linked sub-tasks.

File: docs/source/development/style-guide/review-standards.rst
Comment (sdhoyt) [inline docs/source/development/style-guide/review-standards.rst:56]: Should we set a rule of thumb for how long you should keep a PR open to give people time to review? Like 5 working days unless it's time sensitive or you get confirmation that no one else is planning to review?

**Comment by GFMoraga** [comment]
Looks great! The only confusion I had was "marking the pull request as wip", The link works, but it leads me to draft PR. Which I think it is great, but is a 'wip' the same as 'draft'?

**Comment by sdhoyt** [comment]
Great job on this! I think this will be really helpful

**Review by GFMoraga — COMMENTED** [review]
Very helpful!

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst]: Ok, I do like that. I will add a note about specific requirements. I'd also be ok with requiring an issue period unless it's a documentation/ github workflow/pre commit PR, but I didn't want to make that change without discussing it with the team

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst:56]: I do like that, although I think 5 is a little long. Maybe 3? It would be good to add a note that if you don't get any reviews after 3-4 days, to ping a reminder in the PR or in the slack channel, or ask the person who is closest to the issue/best reviewer.

**Comment by maxinelasp** [comment]
> Looks great! The only confusion I had was "marking the pull request as wip", The link works, but it leads me to draft PR. Which I think it is great, but is a 'wip' the same as 'draft'?

Yes it is, thank you for calling that out. I will fix that confusion.

File: docs/source/development/style-guide/review-standards.rst
Comment (sdhoyt) [reply docs/source/development/style-guide/review-standards.rst:56]: Yeah 3 would be good. I just want to avoid a situation where someone gets a review and approval on the same day they open the PR, then merge it right away before others have a chance to look, so 3 days sounds good too.

File: docs/source/development/style-guide/review-standards.rst
Comment (bourque) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
#. Comments should be polite, but straightforward and candid
```

**Review by bourque — APPROVED** [review]
I agree with Sean and Gabriel -- this is great and very helpful to have!

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
Before any code is merged into the code base, it will need to be put up for pull request (PR) and reviewed. The pull request should be created following
```

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
Before opening a pull request
```

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
the proposed changes meet the repository standards.
```

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
However, all code merged into the ``dev`` branch should meet the repository standards and work as expected.
```

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]: Could you also give a description for how to link the PR or link to the documentation for how to add the specific keywords to close issues automatically?
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword#linking-a-pull-request-to-an-issue-using-a-keyword

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
#. If you make a "request changes" comment, you must follow up and dismiss your "request changes" review once the author has addressed that aspect of the code. Please do this in a timely manner so you do not block the PR for longer than necessary
```

I am hesitant to say that I have to change it to an approving review. I have sometimes wanted to just dismiss my request changes, but still don't feel confident in fully approving a code change. e.g. if I was blocking a small portion like a name change, but I don't have enough expertise in the broader larger code changes to speak for an approval.

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst:57]: This brings up two thoughts.
1. Are we approving too easily right now? Maybe we should save approvals for when we feel like it has addressed all of our comments?

2. Should we change from self-merging to reviewer-merging. i.e. we don't merge our own code, only a reviewer that thinks it is satisfactory would do that?

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst:59]: Nice, I like this description :)

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst:61]:
```suggestion
#. You should ensure ALL checks on the PR pass (tests are passing and no linting errors)
```
Do we want to call out linting specifically? Maybe to jargony though...

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst:62]: Do we want to add something about this in the style guide and not just the PR guide? i.e. something about good self-contained commits and linking out to commit messages. Totally fine to leave this as a follow-up to add over there if you think it is worth it.

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]: Can this whole section be deleted if we follow the "linking PRs to automatically close issues"? And we are on forks, so no big deal for our team about removing branches if other people want to keep their stuff around.

**Review by greglucas — COMMENTED** [review]
Thank you for gathering all of this information, I think it is great! I did have quite a few clarification comments though, sorry for lots of questions as I was going through it.

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst:57]: I think leaving it as self-merging is fine. A reviewer might not know that, for example, I really want a review from someone specific but haven't gotten it yet.

What I was trying to express with this comment is that if Matthew gives me an approval, but Greg has outstanding comments, I should  not merge until I have addressed Greg's comments. 

I usually give an approval as a "when you fix this" type thing. If it's a minor change that I trust the author to make without needing another review, than I will give an approval, but I still want them to make the changes I suggested. In my opinion, even if the PR has approving reviews, the comments should still be addressed (where addressed means either fixed or replied to with why they will not be fixed). 

Do you or others have other opinions? I would worry about dragging out the review process for minor things - I think people won't approve something which they have larger concerns around. But if it's something minor, like suggesting additional documentation, then the reviewer doesn't necessarily need to go back and review a change after it happens. (e.g. if I say "please add a docstring here" then my approval stands for whatever docstring the author adds)

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst:61]: I think it's covered under all checks personally, I mostly added the parentheses as a clarification for what "checks" means. I don't feel strongly about that though, you can commit this suggestion if you think it makes more sense.

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst:62]: I think that is a good suggestion, but out of scope for this document :) I will write up an issue to add that to the style guide.

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst]: Yeah, that's a good point. I will make that change.

**Review by sdhoyt — APPROVED** [review]

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [reply docs/source/development/style-guide/review-standards.rst]: missed this one?

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [reply docs/source/development/style-guide/review-standards.rst:61]: Nope, this is fine with me!

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [reply docs/source/development/style-guide/review-standards.rst:57]: > I usually give an approval as a "when you fix this" type thing. If it's a minor change that I trust the author to make without needing another review, than I will give an approval, but I still want them to make the changes I suggested. In my opinion, even if the PR has approving reviews, the comments should still be addressed (where addressed means either fixed or replied to with why they will not be fixed).

I usually do the same, so I agree with this. This was mostly just a thought of "if we care enough about these things", but it sounds like possibly not, so lets let it organically evolve and only update this if we run into issues in the future.

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [reply docs/source/development/style-guide/review-standards.rst]: @maxinelasp I think you missed some of these nits, or did you not want to accept them? I do sort of think we should avoid looking like dictating that this is "ours" and just go the generic route and talk about abstract "repository/organization" instead of people "us/our"

Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst]: Yes, apparently these commits got erased somehow during my last edit. Let me try and fix that

File: docs/source/development/style-guide/review-standards.rst
Comment (greglucas) [inline docs/source/development/style-guide/review-standards.rst]:
```suggestion
=============================
```
I think this is now too short. It'd be nice if we had the doc-build PRs linked for easy viewing, or maybe just running the sphinx builder with warnings as an extra test so it could log this out?

File: docs/source/development/style-guide/checklist-for-pull-requests.rst
Comment (greglucas) [inline docs/source/development/style-guide/checklist-for-pull-requests.rst]:
```suggestion
Before merging, make sure you have addressed all the comments in the review adhering to the :ref:`PR standards <pull-request-review-standards>`.
```

**Review by greglucas — APPROVED** [review]
Looks great, thanks for following up on the changes! Two minor nits below that you can take or leave.

**Review by laspsandoval — APPROVED** [review]
Nicely done!

File: docs/source/development/style-guide/review-standards.rst
Comment (tech3371) [inline docs/source/development/style-guide/review-standards.rst]: I feel like draft PR means it's not in final state but would like someone to give initial review.

**Review by tech3371 — APPROVED** [review]
Looks good besides one comment.

File: docs/source/development/style-guide/review-standards.rst
Comment (maxinelasp) [reply docs/source/development/style-guide/review-standards.rst]: I added a clarification, does that better line up with what you're thinking?

Comment (tech3371) [reply docs/source/development/style-guide/review-standards.rst]: yes. Thank you!

**Review by bryan-harter — APPROVED** [review]

**Pull request merged**
